test/unit/SourceLocator/Type/FileSystemIteratorSourceLocatorTest.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace BetterReflectionTest\SourceLocator\Type;
+
+use BetterReflection\Reflector\ClassReflector;
+use BetterReflection\SourceLocator\Type\FileSystemIteratorSourceLocator;
+use BetterReflection\SourceLocator\Type\SingleFileSourceLocator;
+
+/**
+ * @covers \BetterReflection\SourceLocator\Type\FileSystemIteratorSourceLocator
+ */
+class FileSystemIteratorSourceLocatorTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var string
+     */
+    private $directoryToScan = __DIR__ . '/../../Assets/DirectoryScannerAssets';
+
+    /**
+     * @var FileSystemIteratorSourceLocator
+     */
+    private $sourceLocator;
+
+    public function setUp()
+    {
+        $fileSystemIterator = new \RecursiveDirectoryIterator($this->directoryToScan, \RecursiveDirectoryIterator::SKIP_DOTS);
```

Comment:
If you type-hint against `Iterator`, you can then pass in an `ArrayIterator`, or similar - much simpler tests :)